crates/libs/windows/src/core/agile_reference.rs

```diff
@@ -9,11 +9,9 @@ pub struct AgileReference<T>(IAgileReference, PhantomData<T>);
 
 impl<T: Interface> AgileReference<T> {
     /// Creates an agile reference to the object.
-    pub fn new<'a>(object: &'a T) -> Result<Self>
-    where
-        &'a T: IntoParam<'a, IUnknown>,
-    {
-        unsafe { RoGetAgileReference(AGILEREFERENCE_DEFAULT, &T::IID, object).map(|reference| Self(reference, Default::default())) }
+    pub fn new(object: &T) -> Result<Self> {
+        let unknown: &IUnknown = unsafe { std::mem::transmute(object) };
+        unsafe { RoGetAgileReference(AGILEREFERENCE_DEFAULT, &T::IID, unknown).map(|reference| Self(reference, Default::default())) }
```

Comment:
This just removes the redundant constraint on the `AgileReference` constructor. The `Interface` trait doesn't "know" that it requires `IUnknown`. That's something I'll try to fix separately.